Unity3D/Assets/RosSharp/Scripts/RosCommuncation/JointStateSubscriber.cs

```diff
@@ -22,15 +22,21 @@ public class JointStateSubscriber : Subscriber<Messages.Sensor.JointState>
         public List<string> JointNames;
         public List<JointStateWriter> JointStateWriters;
 
-		protected override void Start()
-		{
-			base.Start();
-		}
-		
+        protected override void Start()
+        {
+            base.Start();
+        }
+
         protected override void ReceiveMessage(Messages.Sensor.JointState message)
         {
             for (int i = 0; i < message.name.Length; i++)
-                JointStateWriters[ JointNames.IndexOf(message.name[i]) ].Write(message.position[i]);
+            {
+                if (JointNames.Contains(message.name[i]))
+                {
+                    JointStateWriters[JointNames.IndexOf(message.name[i])].Write(message.position[i]);
+                }
+
```

Comment:
Would this solution also work for you?
```
 protected override void ReceiveMessage(Messages.Sensor.JointState message)
 {
            int index;
            for (int i = 0; i < message.name.Length; i++)
            {
                index = JointNames.IndexOf(message.name[i]);
                if (index!= -1)
                    JointStateWriters[index].Write(message.position[i]);
            }
  }
```
With this solution we will not increase the amount of search operations in the JointNames list.
By calling `Contains` and `IndexOf`we search the list twice.